libs/sdk-core/src/input_parser.rs

```diff
@@ -261,17 +261,26 @@ pub(crate) async fn post_and_log_response(url: &str, body: Option<String>) -> Sd
 pub(crate) async fn get_and_log_response(url: &str) -> SdkResult<String> {
     debug!("Making GET request to: {url}");
 
-    let raw_body = get_reqwest_client()?
+    let response = get_reqwest_client()?
         .get(url)
         .send()
         .await
-        .map_err(|e| SdkError::ServiceConnectivity { err: e.to_string() })?
-        .text()
-        .await
         .map_err(|e| SdkError::ServiceConnectivity { err: e.to_string() })?;
-    debug!("Received raw response body: {raw_body}");
-
-    Ok(raw_body)
+    match response.status().is_success() {
+        true => response
+            .text()
+            .await
+            .map_err(|e| SdkError::ServiceConnectivity { err: e.to_string() }),
+        false => {
+            let err = format!(
+                "GET request {} failed with status: {}",
+                url,
+                response.status()
+            );
+            error!("{err}");
```

Comment:
Is it worth logging the response body in unsuccessful requests or is the http code enough to diagnose potential issues?

Reply:
I think it worth since it shouldn't happen too much and when it happens like in the associated issue it helps identify the problem.